src/frontend/src/flows/verifiableCredentials/index.ts

```diff
@@ -85,9 +86,8 @@ const verifyCredentials = async ({
   const issuerCanisterId = lookedUp.ok;
 
   // If the RP provided a canister ID, check that it matches what we got
-  if (nonNullish(expectedIssuerCanisterId_)) {
-    const expectedCanisterId = expectedIssuerCanisterId_?.toText();
-    if (expectedCanisterId !== issuerCanisterId) {
+  if (nonNullish(expectedIssuerCanisterId)) {
+    if (expectedIssuerCanisterId.compareTo(issuerCanisterId) !== "eq") {
```

Comment:
Do you know what would it mean great or less than for principals? Why not just `isEqual` that returns a `boolean`?

Reply:
This is the API that is offered by `agent-js`. Ordering can be useful for certain data structures / applications. But I'm not sure if there was a concrete requirement to support it. The ordering is done by byte sequence.